internal/databases/clickhouse/backup_push_handler.go

```diff
@@ -0,0 +1,91 @@
+package clickhouse
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/wal-g/tracelog"
+	"github.com/wal-g/wal-g/utility"
+	"github.com/wal-g/wal-g/internal"
+)
+
+func HandleBackupPush(uploader *internal.Uploader, backupCmd *exec.Cmd) {
+	backupFolderRootPath, _ := internal.GetSetting(internal.ClickHouseBackupPath)
+
+	backupName := getBackupName()
+	appendBackupNameToCommand(backupCmd, backupName)
+	_, stderr, err := utility.StartCommandWithStdoutStderr(backupCmd)
+	if err := backupCmd.Wait(); err != nil {
+		tracelog.ErrorLogger.Printf("backup command output:\n%s", stderr.String())
+		tracelog.ErrorLogger.Fatalf("failed to run backup: %v", err)
+	}
+
+	backupFolderPath := path.Join(backupFolderRootPath, backupName)
+	filePaths, err := getBackupFilePaths(backupFolderPath)
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("failed to get backup filepaths: %v", err)
+	}
+
+	for _, filePath := range filePaths {
+		err = uploadBackupFile(uploader, backupFolderRootPath, filePath)
```

Comment:
Currently, the file uploading is synchronous, which is slow. Also, the upload process uploads the files one by one, which might be inefficient if we have a lot of files. I suggest utilizing `TarBallQueue` to solve these two problems.

You can see how `TarBallQueue` is being used in Postgres and use it as a reference.